connection_pool/connection_pool_test.go

```diff
@@ -0,0 +1,1379 @@
+package connection_pool_test
+
+import (
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tarantool/go-tarantool"
+	"github.com/tarantool/go-tarantool/connection_pool"
+	"github.com/tarantool/go-tarantool/test_helpers"
+)
+
+var spaceNo = uint32(520)
+var spaceName = "testPool"
+var indexNo = uint32(0)
+var indexName = "pk"
+
+var ports   = []string{"3013", "3014", "3015",
+                       "3016", "3017"}
+var host    = "127.0.0.1"
+var servers = []string{strings.Join([]string{host, ports[0]}, ":"),
+                       strings.Join([]string{host, ports[1]}, ":"),
+                       strings.Join([]string{host, ports[2]}, ":"),
+                       strings.Join([]string{host, ports[3]}, ":"),
+                       strings.Join([]string{host, ports[4]}, ":")}
+
+var connOpts = tarantool.Opts{
+	Timeout: 500 * time.Millisecond,
+	User:    "test",
+	Pass:    "test",
+}
+
+var connOptsPool = connection_pool.OptsPool{
+	CheckTimeout:         1 * time.Second,
+	NodesGetFunctionName: "get_cluster_nodes",
+	ClusterDiscoveryTime: 3 * time.Second,
+}
+
+var instances []test_helpers.TarantoolInstance
+
+func TestConnError_IncorrectParams(t *testing.T) {
+	connPool, err := connection_pool.Connect([]string{}, tarantool.Opts{})
+	if err == nil {
+		t.Errorf("err is nil with incorrect params")
+	}
+	if connPool != nil {
+		t.Errorf("conn is not nil with incorrect params")
+	}
+	if err.Error() != "addrs should not be empty" {
+		t.Errorf("incorrect error: %s", err.Error())
+	}
+
+	connPool, err = connection_pool.Connect([]string{"err1", "err2"}, connOpts)
+	if err == nil {
+		t.Errorf("err is nil with incorrect params")
```

Comment:
Replaced with assert (I use package `require`, because  `require` implements the same assertions as the `assert` package but stops test execution when a test fails.)